### DISCUSSION TITLE
N64 and PCSX controllers

### DISCUSSION THREAD

**Discussion by saxsor** [giongto35/cloud-game#484] [Q&A]
Hi everyone, do you know if there is a way to configure the controllers for n64 games or for pcsx games, since the layout of the controllers are different that the one showned on the actual UI, thanks!!

**Comment by sergystepanov** [comment]
Hey! It is not possible in a user friendly way at the moment. You can change controller mappings in this JS file in this function https://github.com/giongto35/cloud-game/blob/master/web/js/input/joystick.js#L93 for all games.

**Reply by giongto35** [reply]
@saxsor  I don't get your question. What UI you are expecting for pcsx game? Is it with the joystick?
@sergystepanov can we control the mapping by file extension?

**Reply by sergystepanov** [reply]
@giongto35, do you mean adding a config mapping for the Retropad [state](https://github.com/giongto35/cloud-game/blob/master/web/js/input/retropad.js#L12) on the server for cores? What exactly is your idea?

In the past, people always wanted custom gamepad mappings on a per-game basis. :/ On the client side, it is necessary to implement some way to configure the controllers, at least to some extent. :/

**Reply by saxsor** [reply]
Hi, yes it's helpful @sergystepanov , and @giongto35 just like @sergystepanov said, an improvement on the mapping of controllers on the client side would be great, so you can customize it as you need

**Reply by saxsor** [reply]
since at the moment, for example when you plug an xbox controller or other kind of controller, the buttons are scrammbled and when you try to customize them, it's really complicated

**Reply by sergystepanov** [reply]
I'll try to add something when I get my new laptop in January, hopefully. #339
